run.sh

```diff
@@ -1,46 +1,74 @@
 #!/usr/bin/env bash
 
+# Define directories
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 REPODIR="$(dirname "$SCRIPTDIR")"
 ARKSERVER=${ARKSERVER_SHARED:-"/ark/server"}
 
-# always fail script if a cmd fails
+# Always fail script if a command fails
 set -eo pipefail
 
 echo "###########################################################################"
 echo "# Ark Server - " `date`
 echo "###########################################################################"
 
-echo "Ensuring correct permissions..."
-sudo find /ark -not -user steam -o -not -group steam -exec chown -v steam:steam {} \;
-sudo find /home/steam -not -user steam -o -not -group steam -exec chown -v steam:steam {} \;
+# Determine if the user has root or sudo privileges
+if [ "$(id -u)" -eq 0 ]; then
+    HAS_PRIVILEGES=true
```

Comment:
I would shorten this to:
```
HAS_PRIVILEGES=false # default
if [ [ "$(id -u)" -eq 0 ]; then
    HAS_PRIVILEGES=true
fi
```
Why did you add so many different checks for being root?

Reply:
Good point, we just need to check if the user can run sudo.